src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -284,6 +286,119 @@ public bool PrefixReserved
+        }
+
+        public bool PackageDeprecationAlternatePackageRangeHasUpperBound
+        {
+            get { return _packageMetadata?.DeprecationMetadata?.AlternatePackage?.Range?.HasUpperBound ?? false; }
+        }
+
+        private Uri _packageDeprecationAlternatePackageLinkUrl;
+        public Uri PackageDeprecationAlternatePackageLinkUrl
+        {
+            get { return _packageDeprecationAlternatePackageLinkUrl; }
+            set
+            {
+                if (_packageDeprecationAlternatePackageLinkUrl != value)
+                {
+                    _packageDeprecationAlternatePackageLinkUrl = value;
+
+                    OnPropertyChanged(nameof(PackageDeprecationAlternatePackageLinkUrl));
+                }
+            }
+        }
+
+        private string _packageDeprecationReasons;
+        public string PackageDeprecationReasons
+        {
+            get { return _packageDeprecationReasons; }
+            set
+            {
+                if (_packageDeprecationReasons != value)
+                {
+                    _packageDeprecationReasons = value;
+
+                    OnPropertyChanged(nameof(PackageDeprecationReasons));
+                }
+            }
+        }
+
+        private string _packageDeprecationReasonsExplained;
+        public string PackageDeprecationReasonsExplained
+        {
+            get { return _packageDeprecationReasonsExplained; }
+            set
+            {
+                if (_packageDeprecationReasonsExplained != value)
+                {
+                    _packageDeprecationReasonsExplained = value;
+
+                    OnPropertyChanged(nameof(PackageDeprecationReasonsExplained));
+                }
+            }
+        }
+
+        private string _packageDeprecationCustomMessage;
+        public string PackageDeprecationCustomMessage
+        {
+            get { return _packageDeprecationCustomMessage; }
+            set
+            {
+                if (_packageDeprecationCustomMessage != value)
+                {
+                    _packageDeprecationCustomMessage = value;
+                    OnPropertyChanged(nameof(PackageDeprecationCustomMessage));
+                }
+            }
+        }
+
+        private string _packageDeprecationAlternatePackageLinkText;
+        public string PackageDeprecationAlternatePackageLinkText
+        {
+            get { return _packageDeprecationAlternatePackageLinkText; }
+            set
+            {
+                if (_packageDeprecationAlternatePackageLinkText != value)
+                {
+                    _packageDeprecationAlternatePackageLinkText = value;
+
+                    OnPropertyChanged(nameof(PackageDeprecationAlternatePackageLinkText));
+                }
+            }
+        }
+
+        public string ExplainPackageDeprecationReasons(IEnumerable<string> reasons)
+        {
+            if (reasons.Contains(PackageDeprecationReason.CriticalBugs, StringComparer.OrdinalIgnoreCase))
```

Comment:
It's unclear from @scottbommarito's docs whether these strings are case sensitive or not. https://github.com/NuGet/docs.microsoft.com-nuget/pull/1460#issuecomment-504082926

Reply:
Commented on the docs PR as well:

Nothing on the server side parses the list of reasons, but it will always be in exposed in V3 with the same casing. I'm fine with either case-sensitive reason strings or case-insensitive reason strings, but we should add to the docs which we choose.

Reply:
I just thought treating them as case-insensitive seemed like the safest approach on the client-side (e.g. in the unlikely event there's ever an incorrect casing that slips in, older clients wouldn't be able to recognize the change). Should we be more strict and enforce case-sensitivity?